src/server/context.ts

```diff
@@ -359,6 +362,30 @@ export class ServerContext {
 
     const dev = isDevMode();
     if (dev) {
+      const checks: CheckFunction[] = [
+        () => assertModuleExportsDefault(app, "_app"),
+        () => assertSingleModule(routes, "_app"),
+        () => assertModuleExportsDefault(unknownModule, "_404"),
+        () => assertSingleModule(routes, "_404"),
+        () => assertModuleExportsDefault(errorModule, "_500"),
+        () => assertSingleModule(routes, "_500"),
+        () => assertRoutesHaveHandlerOrComponent(routes),
+        () => assertStaticDirSafety(opts.staticDir ?? "", defaultStaticDir),
+        () => assertNoStaticRouteConflicts(routes, staticFiles),
+        () => assertPluginsCallRender(opts.plugins ?? []),
```

Comment:
apparently these checks currently cause the following issue in the ci:
```sh
error: Leaking async ops:
  - 1 async operation to sleep for a duration was started before this test, but was completed during the test. Async operations should not complete in a test if they were not started in that test.
            This is often caused by not cancelling a `setTimeout` or `setInterval` call.
```

would it make sense to add yet another check for testing in `isDevMode`?